src/masternodes/mn_checks.cpp

```diff
@@ -3984,15 +3984,18 @@ Res HasAuth(const CTransaction &tx, const CCoinsViewCache &coins, const CScript
                 auto it = input.scriptSig.begin();
                 CPubKey pubkey(input.scriptSig.begin() + *it + 2, input.scriptSig.end());
                 if (pubkey.Decompress()) {
-                    auto script = GetScriptForDestination(WitnessV16EthHash(pubkey));
-                    if (script == auth)
+                    const auto script = GetScriptForDestination(WitnessV16EthHash(pubkey));
+                    const auto scriptOut = GetScriptForDestination(PKHash(pubkey));
+                    if (script == auth && coin.out.scriptPubKey == scriptOut)
                         return Res::Ok();
                 }
-            } else if (solution == txnouttype::TX_WITNESS_V0_KEYHASH) {
+            } else if (solution == txnouttype::TX_WITNESS_V0_KEYHASH &&
```

Comment:
We don't need the stack.size() == 2, since the Solver already verifies this, and we also do the reverse conversion check now